drivers/sensor/lis2mdl/lis2mdl.c

```diff
@@ -179,6 +184,18 @@ static int lis2mdl_sample_fetch_mag(const struct device *dev)
 	struct lis2mdl_data *lis2mdl = dev->data;
 	int16_t raw_mag[3];
 
+#ifdef CONFIG_LIS2MDL_SINGLE_MODE
+	if (lis2mdl_operating_mode_set(lis2mdl->ctx, LIS2MDL_SINGLE_TRIGGER)) {
+		LOG_DBG("set single mode failed\n");
```

Comment:
I think that here should be LOG_ERR(). 
I know that the code is full of LOG_DBG() in same type of error cases. I guess we need to fix it sooner or later.